### ISSUE TITLE
Close PRs of obsolete updates

### ISSUE THREAD

**Issue by fthomas** [scala-steward-org/scala-steward#119]
As suggested by @fwbrasil in https://github.com/getquill/quill/pull/1195#issuecomment-433567944.

**Comment by kiranbayram** [comment]
Have you had a solution idea for this? @fthomas

**Comment by mslinn** [comment]
This is a good idea. For example, the AWS OSGI Java library puts out a revision every few days, and they really pile up. The current queue of PRs for my [awslib_scala project](https://github.com/mslinn/awslib_scala) is  output by a [bash script that I wrote to apply PRs](https://gist.github.com/mslinn/1caab61a8222dfc25f6d9df5989af818) called `prs`:
```
     #47  Update sbt to 1.3.7
     #46  Update aws-java-sdk-osgi to 1.11.707
     #45  Update aws-java-sdk-osgi to 1.11.706
     #44  Update httpcore to 4.4.13
     #42  Update aws-java-sdk-osgi to 1.11.704
     #41  Update aws-java-sdk-osgi to 1.11.703
     #40  Update aws-java-sdk-osgi to 1.11.702
     #39  Update aws-java-sdk-osgi to 1.11.701
     #37  Update commons-codec to 1.14
```
Until Scala Steward incorporates this feature request, I think I'll modify my `prs` script to:
 - Scan the output of the `hub pr list` command
 - For each PR:
    - If obsolete, delete the PR, otherwise apply it

**Comment by daddykotex** [comment]
Just an update, I'll probably tackle this.

This is something really interesting when running Scala Steward within an organization. My company has a lot of libraries in independent repositories and they're updated frequently. Along with this, when we run a build storm, it produces a lot of new libraries.

Overtime, if we're not disciplined and merge quickly they add up and it's hard to maintain.

This issue is also very interesting to me but the problems @fthomas raises are real, so I'll wait out for it.

https://github.com/scala-steward-org/scala-steward/issues/412

**Comment by daddykotex** [comment]
I'm done with the implementation and most of the testing, I just waiting a new version of https://github.com/aws/aws-sdk-java/tags to test all the way :P

Ideally, it should close: https://gitlab.com/daddykotex/scala-steward-test/-/merge_requests/7

**Comment by daddykotex** [comment]
It worked (but I'll be honest, on the second try)

```
[info] 2020-10-15 20:08:26,465 INFO
[info]   ____            _         ____  _                             _
[info]  / ___|  ___ __ _| | __ _  / ___|| |_ _____      ____ _ _ __ __| |
[info]  \___ \ / __/ _` | |/ _` | \___ \| __/ _ \ \ /\ / / _` | '__/ _` |
[info]   ___) | (_| (_| | | (_| |  ___) | ||  __/\ V  V / (_| | | | (_| |
[info]  |____/ \___\__,_|_|\__,_| |____/ \__\___| \_/\_/ \__,_|_|  \__,_|
[info]  v0.7.0-40-0ba43553-20201015-1905-SNAPSHOT
[info]
[info] 2020-10-15 20:08:27,695 INFO  Loaded 15 Scalafix migrations
[info] 2020-10-15 20:08:27,859 INFO  Run self checks
[info] 2020-10-15 20:08:28,733 INFO  Clean workspace /Users/david/mnubo/repos/dataviz/scala-steward/run
[info] 2020-10-15 20:08:28,757 INFO  Add global sbt plugins
[info] 2020-10-15 20:08:28,814 INFO  ──────────── Steward daddykotex/scala-steward-test ────────────
[info] 2020-10-15 20:08:28,816 INFO  Check cache of daddykotex/scala-steward-test
[info] 2020-10-15 20:08:29,569 INFO  Find updates for daddykotex/scala-steward-test
[info] 2020-10-15 20:08:29,807 INFO  Ignore org.typelevel:(cats-effect, cats-effect_2.13) : 2.2.0 -> 3.0-ac83d13 -> 3.0-bb20d55 -> ... -> 3.0-805b021 -> 3.0-8006968 -> 3.0-8096649 (reason: no suitable next version)
[info] 2020-10-15 20:08:30,198 INFO  Found 1 update:
[info]   com.amazonaws:aws-java-sdk-s3 : 1.11.870 -> 1.11.881
[info] 2020-10-15 20:08:30,205 INFO  daddykotex/scala-steward-test is outdated:
[info]   new version: com.amazonaws:aws-java-sdk-s3 : 1.11.870 -> 1.11.881
[info] 2020-10-15 20:08:30,206 INFO  Nurture daddykotex/scala-steward-test
[info] 2020-10-15 20:08:30,207 INFO  Clone and synchronize daddykotex/scala-steward-test
[info] 2020-10-15 20:08:30,942 INFO  Found 1 update:
[info]   com.amazonaws:aws-java-sdk-s3 : 1.11.870 -> 1.11.881
[info] 2020-10-15 20:08:30,961 INFO  Process update com.amazonaws:aws-java-sdk-s3 : 1.11.870 -> 1.11.881
[info] 2020-10-15 20:08:31,216 INFO  Trying heuristic 'moduleId'
[info] 2020-10-15 20:08:31,266 INFO  Create branch update/aws-java-sdk-s3-1.11.881
[info] 2020-10-15 20:08:31,333 INFO  Push 1 commit(s)
[info] 2020-10-15 20:08:32,428 INFO  Create PR update/aws-java-sdk-s3-1.11.881
[info] 2020-10-15 20:08:33,821 INFO  Created PR https://gitlab.com/daddykotex/scala-steward-test/-/merge_requests/12
[info] 2020-10-15 20:08:33,846 INFO  Looking to close obsolete PRs for aws-java-sdk-s3
[info] 2020-10-15 20:08:34,193 INFO  Closed a PR for com.amazonaws:aws-java-sdk-s3 : 1.11.870 -> 1.11.880
[info] 2020-10-15 20:08:34,230 INFO  ──────────── Total time: Steward daddykotex/scala-steward-test: 5s 415ms ────────────
[info] 2020-10-15 20:08:34,233 INFO  ──────────── Total time: run: 6s 376ms ────────────
[success] Total time: 14 s, completed 15-Oct-2020 8:08:37 PM
```

**Comment by fthomas** [comment]
Implemented in #1667.
